File: sdk/runanywhere-swift/Sources/RunAnywhere/Public/Configuration/RoutingPolicy.swift
import Foundation

/// Routing policy determines how requests are routed between device and cloud
public enum RoutingPolicy: String, Codable {
    /// Automatically determine best execution target
    case automatic

    /// Always use on-device execution when possible
    case preferDevice

    /// ONLY use on-device execution - never use cloud
    case deviceOnly

    /// Always use cloud execution
    case preferCloud

    /// Use custom routing rules
    case custom
}

File: sdk/runanywhere-android/src/main/java/com/runanywhere/sdk/configuration/RoutingPolicy.kt
package com.runanywhere.sdk.configuration

/**
 * Routing policy for model selection
 */
enum class RoutingPolicy {
    /**
     * Automatic routing based on device capabilities and model requirements
     */
    AUTOMATIC,
    
    /**
     * Always prefer on-device execution
     */
    ON_DEVICE_ONLY,
    
    /**
     * Always prefer cloud execution
     */
    CLOUD_ONLY,
    
    /**
     * Hybrid routing with fallback
     */
    HYBRID
} 